client/src/javascript/components/modals/feeds-modal/DownloadRulesTab.js

```diff
@@ -111,8 +130,36 @@ class DownloadRulesTab extends React.Component {
     }
   }, 150);
 
+  checkMatch(match, exclude, check) {
+    let checkMatchTextbox;
+    for (let i = 0; i < this.formRef.formRef.length; i++) {
+      let element = this.formRef.formRef[i];
+      if (element.name === 'check') {
+        checkMatchTextbox = element;
+      }
+    }
+
+    possibleMatchClasses.forEach(className => {
+      checkMatchTextbox.classList.remove(className);
+    });
+
+    if (!Validator.isNotEmpty(match) || !Validator.isRegExValid(match) || !Validator.isRegExValid(exclude)) {
+      return;
+    }
+
+    const isMatched = new RegExp(match, 'gi').test(check);
+    const isExcluded = exclude !== '' && new RegExp(exclude, 'gi').test(check);
+
+    if (isMatched && !isExcluded) {
+      checkMatchTextbox.classList.add('check-match-success');
+    } else {
+      checkMatchTextbox.classList.add('check-match-failed');
```

Comment:
i know this goes against your standards but why avoid a class name state manager that's bound into vanilla js and glue css classes together in a stringified form?  
i'd say using the classList api is in general better practice. though i agree this code could be better. instead of `remove` and `add` one could use `toggle` and provide the conditional expression as parameter to it.

Reply:
When I added a className property to the checkMatchTextbox component, that class did not show up in the resulting HTML. I think this is caused by the fact that checkMatchTextbox refers to a component which comes from flood-ui-kit, rather than a standard HTML tag.

So the classList approach was the only way i could think of to manipulate the class of this component.

Reply:
@GottZ Because this approach breaks React's DOM abstraction.

@jwbuiter If we need to provide a custom `className` to a `flood-ui-kit` component, then the component should be updated in `flood-ui-kit`